### PR TITLE
Feature/card option logic

### PR DESCRIPTION
**Pull request by kcbreuil**
WIP

### CODE REVIEW

File: studio/schemas/documents/backgroundImgwithCard.js [hunk 1]
```diff
@@ -18,5 +18,54 @@ export default {
...
+        {
+            name: 'button',
+            title: 'Button',
+            type: 'string',
+            description: 'Selection will give card a button',
+            options: {
+                list: [{ title: 'hasButton', value: 'hasButton' }],
+                layout: 'radio',
+            },
+        },
```
Comment (asasmith) [inline studio/schemas/documents/backgroundImgwithCard.js:69]: Will these always be internal links? If so we'll need a way to pass a string for the url. We can probably forgo the `button` field and add one for the url. Should be able to evaluate that field (an empty string evaluates to `false`) to control button/cta display logic.

**Review by asasmith — CHANGES_REQUESTED** [review]

File: studio/schemas/documents/backgroundImgwithCard.js [hunk 2]
```diff
@@ -18,5 +18,73 @@ export default {
...
+        {
+            name: 'button',
+            title: 'Button',
+            type: 'string',
+            description: 'Selection will give card a button',
+            options: {
+                list: [{ title: 'Button', value: 'button' }],
+            },
+        },
```
Comment (asasmith) [inline studio/schemas/documents/backgroundImgwithCard.js:56]: I don't think you need this field

File: web/src/components/bgImagewithCard.js [hunk 3]
```diff
@@ -7,25 +7,29 @@ import { convertToBgImage } from 'gbimage-bridge';
...
 
+    const isDark = backgroundColor === 'dark';
+    const isTransparent = backgroundColor === 'transparent';
+    const isRight = cardOrientation === 'right';
+    const hasButton = button === 'button';
+    const isPrimary = cardType === 'primary';
+    const isPrimaryDark = cardType === 'secondary';
+    const isSecondaryDark = cardType === 'transparent';
+    const externalLink = linkType === 'external';
+    const internalLink = linkType === 'internal';
```
Comment (asasmith) [inline web/src/components/bgImagewithCard.js:31]: I'd like to see these cleaned up. I don't think we need `isPrimaryDark` and `isSecondaryDark` options. `hasButton` can be evaluated from the `link` field.

File: web/src/components/bgImagewithCard.js [hunk 4]
```diff
@@ -37,33 +41,45 @@ export default function BgImagewithCard({
...
+    isPrimary: false,
+    isPrimaryDark: false,
+    isSecondary: false,
+    isSecondaryDark: false,
+    isTransparent: false,
+    hasButton: false,
+    isDark: false,
+    isRight: false,
+    internalLink: false,
+    externalLink: false,
```
Comment (asasmith) [inline web/src/components/bgImagewithCard.js:66]: Some of these should be true (I think `isPrimary` and `isRight`)

File: web/src/components/button.js [hunk 5]
```diff
@@ -24,18 +25,18 @@ export default function Button({
         isSecondary ? secondaryStyle : ''
     }${isSecondaryDark ? secondaryDarkStyle : ''}`;
 
-    if (to) {
+    if (internalLink) {
         return (
-            <Link to={to} className={finalStyles}>
+            <Link to={`${link}`} className={finalStyles}>
```
Comment (asasmith) [inline web/src/components/button.js:30]: I'd add the leading `/` here that way we aren't relying on the user to get the string just right
```
to={`/${link}`}
```

**Review by asasmith — CHANGES_REQUESTED** [review]

File: web/src/components/bgImagewithCard.js (hunk 3 above)
Comment (kcbreuil) [reply web/src/components/bgImagewithCard.js:31]: yes we do, those are the colors we need, per our convo

Comment (asasmith) [reply web/src/components/bgImagewithCard.js:31]: `isPrimary/isSecondary` and `isLight/isDark` aren't enough?

Comment (kcbreuil) [reply web/src/components/bgImagewithCard.js:31]: is primary and issecondary have to do with button colors. we need to add for transparent (if user wants transparent button). technically its called secondaryDark but if you want me to rename it to transparent button happy to do so but yes I think its necessary.

Comment (asasmith) [reply web/src/components/bgImagewithCard.js:31]: I'm not sure I'm following. I think we should be able to derive all of the stying we need with less props. I think all we need are `isPrimary`, `isSecondary`, `isTransparent`, `isDark`. Adding props like `isPrimaryDark` is superfluous.

Comment (asasmith) [reply web/src/components/bgImagewithCard.js:31]: What prop determines the background color of the card?

Comment (kcbreuil) [reply web/src/components/bgImagewithCard.js:31]: It's not superfluous it is the button for the transparent card.

Comment (kcbreuil) [reply web/src/components/bgImagewithCard.js:31]: isDark determines the color for the card.

File: studio/schemas/documents/backgroundImgwithCard.js (hunk 2 above)
Comment (kcbreuil) [reply studio/schemas/documents/backgroundImgwithCard.js:56]: I think we do, hasButton tells the component it needs a button. {hasButton &&....)

File: web/src/components/bgImagewithCard.js (hunk 3 above)
Comment (asasmith) [reply web/src/components/bgImagewithCard.js:31]: Do we need `isPrimaryDark` too though? That's the question I'm asking.

File: studio/schemas/documents/backgroundImgwithCard.js (hunk 2 above)
Comment (asasmith) [reply studio/schemas/documents/backgroundImgwithCard.js:56]: You could derive a `hasButton` variable based on some other state that is returned from the query. You don't need to have a `button` field in sanity as well.

File: web/src/components/bgImagewithCard.js (hunk 3 above)
Comment (kcbreuil) [reply web/src/components/bgImagewithCard.js:31]: yeah, we are assigning that color scheme - primary dark from the mock ups as the secondary button color --dark card with light button

File: web/src/components/bgImagewithCard.js (hunk 4 above)
Comment (kcbreuil) [reply web/src/components/bgImagewithCard.js:66]: I don't think so tbh. We have defaults (left, and no button at all) and we want to user to make them true to change the color of the button or align it to the right (left is default)

File: web/src/components/bgImagewithCard.js (hunk 3 above)
Comment (asasmith) [reply web/src/components/bgImagewithCard.js:31]: I guess what I'm asking is, can we apply styling if we only have `isPrimary` and `isDark` props?. Do we have to have unique `isPrimaryDark` etc? 

IMHO, the props/css as it is now is very confusing to me. This seems like a good opportunity to refactor to something simpler that will hopefully help future devs.

File: web/src/components/bgImagewithCard.js (hunk 4 above)
Comment (asasmith) [reply web/src/components/bgImagewithCard.js:66]: So what happens if a user creates a card and doesn't select anything?

File: web/src/components/bgImagewithCard.js (hunk 3 above)
Comment (kcbreuil) [reply web/src/components/bgImagewithCard.js:31]: We have a card that can be dark light or transparent. This is set with isDark or isTransparent. Then the card has it's adjoining button, primary (set with the primarybutton) secondary (set with isprimarydark) or transparent (set with secondaryDark). We spoke about this yesterday needing the third button for the transparent card. I can flesh out the description and write up some documentation to make this more clear?

File: web/src/components/bgImagewithCard.js (hunk 4 above)
Comment (kcbreuil) [reply web/src/components/bgImagewithCard.js:66]: it will be left, and there is no button color

Comment (asasmith) [reply web/src/components/bgImagewithCard.js:66]: Where is that defined?

**Pull request merged**